cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -176,6 +179,41 @@ func (p *BicepProvider) EnsureEnv(ctx context.Context) error {
 	return nil
 }
 
+func locationParameterFilterImpl(param azure.ArmTemplateParameterDefinition, location account.Location) bool {
+	if param.AllowedValues == nil {
+		return true
+	}
+
+	return slices.IndexFunc(*param.AllowedValues, func(v any) bool {
+		s, ok := v.(string)
+		return ok && location.Name == s
+	}) != -1
+}
+
+// defaultLocationToSelectFn resolves if there is an intention from a location parameter to use a default location.
+// If the parameter has the @allowedValues set, it will return the first value from the array as the default.
+// If the parameter has the location azd metadata, it will check if there is configuration to set a default location.
+// If both @allowedValues and azd location metadata config are set, the metadata config will take precedence.
+// If nothing is set, it will return nil.
+func defaultLocationToSelectFn(locationParam azure.ArmTemplateParameterDefinition) *string {
```

Comment:
Curious about the intention here, I'm not immediately familiar with the suffix "fn" used here.

Reply:
Does it look any different set `defaultPromptValue`

Reply:
taken